File: commons/authorization-commons/src/main/java/com/sv/youapp/common/authorization/dto/UserDTO.java
package com.sv.youapp.common.authorization.dto;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.security.core.CredentialsContainer;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.time.Instant;
import java.util.Set;

@Getter
@ToString
@AllArgsConstructor
public class UserDTO implements UserDetails, CredentialsContainer {

    private final Integer id;
    private final String username;
    @Setter
    private String password;
    private final String email;
    private final String profilePictureUrl;
    private final Instant registeredAt;
    private final Set<GrantedAuthority> authorities;

    @JsonAlias("isEnabled")
    private final boolean enabled;

    @Override
    public boolean isEnabled() {
        return enabled;
    }

    @Override
    public Set<GrantedAuthority> getAuthorities() {
        return authorities;
    }

    @JsonIgnore
    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @JsonIgnore
    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @JsonIgnore
    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public void eraseCredentials() {
        this.password = null;
    }
}

File: authorization-server/src/main/kotlin/com/sv/youapp/authorization/dto/UserDTO.kt
package com.sv.youapp.authorization.dto

import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.time.Instant

class  UserDTO(
    val id: Integer,
    private val username: String,
    private val password: String?,
    val email: String,
    val profilePictureUrl: String?,
    val registeredAt: Instant,
    private val authorities: Set<GrantedAuthority>,
    @JsonAlias("isEnabled")
    private val enabled: Boolean,
) : UserDetails {
    override fun isEnabled() = enabled

    override fun getUsername(): String = username

    override fun getPassword(): String? = password

    override fun getAuthorities() = authorities

    @JsonIgnore
    override fun isAccountNonExpired() = true

    @JsonIgnore
    override fun isAccountNonLocked() = true

    @JsonIgnore
    override fun isCredentialsNonExpired() = true
}
